conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptionHandlerTest.java

```diff
@@ -80,6 +85,59 @@ public void handlesServiceException() throws IOException {
                 .contains("\"parameters\":{\"foo\":\"bar\"}}");
     }
 
+    @Test
+    public void checkServiceExceptionArgumentsSerializedWithToString() throws IOException {
+        record Argument(
+                String field1, Optional<Integer> maybeField2, List<String> collectionField, String nullString) {}
+        exception = new ServiceException(
+                ErrorType.CONFLICT,
+                SafeArg.of("arg", new Argument("foo", Optional.of(42), List.of("bar", "baz"), null)));
+        HttpURLConnection connection = execute();
+
+        assertThat(connection.getResponseCode()).isEqualTo(ErrorType.CONFLICT.httpErrorCode());
+        String expectedSerializedArg =
+                "\"Argument[field1=foo, maybeField2=Optional[42], collectionField=[bar, baz], nullString=null]\"";
+        assertThat(getErrorBody(connection))
+                .contains("{\"errorCode\":\"CONFLICT\"")
+                .contains("\"parameters\":{\"arg\":" + expectedSerializedArg + "}}");
+    }
+
+    @Test
+    public void handlesCheckedServiceException() throws IOException {
```

Comment:
Adding more tests with interesting parameter types: `Optional`, `Collections` and combinations of the two.

Reply:
The piece I'm most interested in testing is `SafeArg.of("emptyOptional", Optional.empty()), SafeArg.of("null", null));`